modules/pointcloudio/src/coders_ply.cpp

```diff
@@ -0,0 +1,293 @@
+#include "coders_ply.hpp"
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sstream>
+
+namespace cv
+{
+namespace pc
+{
+
+std::vector<std::string> split(const std::string& s, char delimiter);
+bool isLittleEndian();
+
+PlyDecoder::PlyDecoder()
+{
+    m_vertices_size = 0;
+}
+
+PlyDecoder::~PlyDecoder()
+{
+
+}
+
+void PlyDecoder::readData( std::vector<Point3f>& points, std::vector<Point3f>& normals )
+{
+    std::ifstream file(m_filename, std::ios::binary);
+    parseHeader(file);
+
+    parseBody(file, points, normals);
+}
+
+inline bool startsWith(const std::string& s1, const std::string& s2) {
+    return s1.compare(0, s2.length(), s2) == 0;
+}
+
+inline std::string trimSpaces(const std::string& input) {
+    size_t start = 0;
+    while (start < input.size() && input[start] == ' ') start++;
+    size_t end = input.size();
+    while (end > start && (input[end - 1] == ' ' || input[end - 1] == '\n' || input[end - 1] == '\r')) end--;
+    return input.substr(start, end - start);
+}
+
+std::vector<std::string> split(const std::string& s, char delimiter)
+{
+   std::vector<std::string> tokens;
+   std::string token;
+   std::istringstream tokenStream(s);
+   while (std::getline(tokenStream, token, delimiter))
+   {
+      tokens.push_back(token);
+   }
+   return tokens;
+}
+
+void PlyDecoder::parseHeader(std::ifstream& file)
+{
+    std::string s;
+    std::getline(file, s);
+    if (trimSpaces(s) != "ply")
+    {
+        std::cout << "Fail isnt ply" << std::endl;
+        return;
```

Comment:
Looks very strange:
- parseHeader should return status to not run parseBody for invalid file
- `std::cout` is not available on Android, iOS and should be be used here. There are `CV_Error` and `CV_Assert` for exceptions and logging API for warnings: https://github.com/opencv/opencv/blob/master/modules/core/include/opencv2/core/utils/logger.hpp